### PR TITLE
bug: properly validating request params for contains content items

### PR DESCRIPTION
**Pull request by alex-sheehan-edx**
## Description

Validating customer uuid in the `https://enterprise-catalog.edx.org/api/v1/enterprise-customer/<CUSTOMER UUID>/contains_content_items/` endpoint

## Ticket Link

https://opsg.in/a/i/2u-internal/64283c70-eb01-407d-b5ba-c33989aad0e8-1674529020487

## Post-review

* [ ] Squash commits into discrete sets of changes
* [ ] Ensure that once the changes have been [deployed to stage](https://gocd.tools.edx.org/go/pipeline/activity/stage-enterprise_catalog), prod is manually deployed

### CODE REVIEW

File: enterprise_catalog/apps/api/v1/views/enterprise_customer.py [hunk 1]
```diff
@@ -84,7 +85,18 @@ def contains_content_items(self, request, enterprise_uuid, course_run_ids, progr
...
-        customer_catalogs = EnterpriseCatalog.objects.filter(enterprise_uuid=enterprise_uuid)
+        try:
+            customer_catalogs = EnterpriseCatalog.objects.filter(enterprise_uuid=enterprise_uuid)
+        except ValueError as exc:
+            logger.warning(
+                f"Could not parse catalogs from provided enterprise uuid: {enterprise_uuid}. "
+                f"Query failed with exception: {exc}"
+            )
+            return Response(
+                f'Error: invalid enterprice customer uuid: {enterprise_uuid} provided.',
```
Comment (johnnagro) [inline enterprise_catalog/apps/api/v1/views/enterprise_customer.py]: maybe put quotes inside the string around `{enterprise_uuid}` to capture/make-obvious whitespace, etc.

File: enterprise_catalog/apps/api/v1/views/enterprise_customer.py [hunk 2]
```diff
@@ -84,7 +85,18 @@ def contains_content_items(self, request, enterprise_uuid, course_run_ids, progr
...
+                status=HTTP_400_BAD_REQUEST
```
Comment (johnnagro) [inline enterprise_catalog/apps/api/v1/views/enterprise_customer.py:98]: what are the callers getting now for a response? a 200 with an empty list? will throwing a 400 start causing weird upstream issues? (do we need to let anyone know, or offer some inbetween to ease the transition)

File: enterprise_catalog/apps/api/v1/views/enterprise_customer.py [hunk 3]
```diff
@@ -84,7 +85,18 @@ def contains_content_items(self, request, enterprise_uuid, course_run_ids, progr
         get_catalog_list = request.GET.get('get_catalog_list', False)
         course_run_ids = unquote_course_keys(course_run_ids)
 
-        customer_catalogs = EnterpriseCatalog.objects.filter(enterprise_uuid=enterprise_uuid)
+        try:
+            customer_catalogs = EnterpriseCatalog.objects.filter(enterprise_uuid=enterprise_uuid)
```
Comment (johnnagro) [inline enterprise_catalog/apps/api/v1/views/enterprise_customer.py]: perhaps we should validate that its a properly formed UUID before we try to use it on the DB?

**Review by johnnagro — APPROVED** [review]
looks good - i have some suggestions to perhaps make it better

File: enterprise_catalog/apps/api/v1/views/enterprise_customer.py (hunk 2 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api/v1/views/enterprise_customer.py:98]: the callers are getting a `500` AFAIK this is breaking on both prod and stage, so in my opinion any change here is a good change.

File: enterprise_catalog/apps/api/v1/views/enterprise_customer.py (hunk 3 above)
Comment (alex-sheehan-edx) [reply enterprise_catalog/apps/api/v1/views/enterprise_customer.py]: sure I think that's fine but doesn't this accomplish essential the same thing?

**Pull request merged**